massa-pos-exports/src/deferred_credits.rs

```diff
@@ -53,6 +54,27 @@ impl DeferredCredits {
             self.0.remove(&slot);
         }
     }
+
+    /// Try to remove a given amount of deferred credit for the given Slot & Address
+    pub fn sub_amount(&mut self, slot: &Slot, address: &Address, amount: &Amount) -> Option<Amount> {
+        let mut result: Option<Amount> = None;
+
+        if self.0.contains_key(slot) {
+            self.0
+                .entry(*slot)
+                .and_modify(|current_credits| {
+                    current_credits
+                        .entry(*address)
+                        .and_modify(|e| {
+                            let sub_amount = e.saturating_sub(*amount);
+                            result = Some(min(*e, *amount));
```

Comment:
I'm not sure I understand the behaviour of this return value, min of the address' credit amount at the given slot and the amount to be removed? In what case is that needed?

Reply:
This can happen if we try to remove more than what is available. 

Let say we want to remove 43 credits when 42 are available; at the end we return 42.